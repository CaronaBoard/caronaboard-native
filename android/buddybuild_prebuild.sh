#!/usr/bin/env bash

ENVFILE="${BUDDYBUILD_SECURE_FILES}/.env"
