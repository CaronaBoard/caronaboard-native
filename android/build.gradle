// Top-level build file where you can add configuration options common to all sub-projects/modules.
ext {
    compileSdkVersion = 26
    buildToolsVersion = '26.0.1'
}

subprojects { subproject ->
    afterEvaluate{
        if((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion
                buildToolsVersion rootProject.ext.buildToolsVersion
            }
        }
    }
}

buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.3.3'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        jcenter()
        maven {
            url 'https://maven.google.com'
            // Alternative URL is 'https://dl.google.com/dl/android/maven2/'
        }
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}

task wrapper(type: Wrapper) {
   gradleVersion = '3.3'
}

task clean(type: Delete) {
  delete rootProject.buildDir
}
