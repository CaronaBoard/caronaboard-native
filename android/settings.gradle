rootProject.name = 'caronaboardnative'
include ':react-native-config'
project(':react-native-config').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-config/android')
include ':react-native-linear-gradient'
project(':react-native-linear-gradient').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-linear-gradient/android')
include ':react-native-vector-icons'
project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
include ':react-native-navigation'
project(':react-native-navigation').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-navigation/android/app/')

include ':app'
