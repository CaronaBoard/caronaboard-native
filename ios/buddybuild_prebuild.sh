#!/usr/bin/env bash

export ENVFILE="${BUDDYBUILD_SECURE_FILES}/.env"
echo "exported ENFILE to ${ENVFILE}."
echo "echoing files in that folder"

for file in ${BUDDYBUILD_SECURE_FILES}/*; do
  echo ${file##*/}
done
