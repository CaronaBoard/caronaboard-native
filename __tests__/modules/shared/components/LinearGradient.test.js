import React from 'react'
import { View } from 'react-native'
import { shallow } from 'enzyme'
import { LinearGradient as iOSLinearGradient } from '../../../../src/modules/shared/components/LinearGradient.ios'
import { LinearGradient as AndroidLinearGradient } from '../../../../src/modules/shared/components/LinearGradient.android'

const props = {
  style: {},
  colors: 'blue',
  start: {x: 0.0, y: 0.5},
  end: {x: 0.0, y: 0.5},
  children: <View />
}

describe('<LinearGradient />', () => {
  describe('iOS', () => {
    it('Snapshot have a snapshot', () => {
      const wrapper = shallow(<iOSLinearGradient {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  // https://github.com/CaronaBoard/caronaboard-native/issues/27
  describe('Android', () => {
    it('Snapshot have a snapshot', () => {
      const wrapper = shallow(<AndroidLinearGradient {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
