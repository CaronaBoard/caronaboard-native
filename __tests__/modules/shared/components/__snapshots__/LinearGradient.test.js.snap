// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinearGradient /> Android Snapshot have a snapshot 1`] = `
<View
  style={Object {}}
>
  <View />
</View>
`;

exports[`<LinearGradient /> iOS Snapshot have a snapshot 1`] = `
<iOSLinearGradient
  colors="blue"
  end={
    Object {
      "x": 0,
      "y": 0.5,
    }
  }
  start={
    Object {
      "x": 0,
      "y": 0.5,
    }
  }
  style={Object {}}
>
  <View />
</iOSLinearGradient>
`;
